### PR TITLE
DTSPB-1519 Add stopped date if case is stopped when updating task list

### PR DESCRIPTION
**Pull request by itoulou**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSPB-1519

### Change description ###
The caseworker actioned the QA step after the Mark as ready to issue but failed the QA step. This leads to the case being stopped. Just like a stopped case, the date when the case was stopped does not show and instead the word Unknown is displayed.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: ccdImports/configFiles/CCD_Probate_Backoffice/CaseEvent.json [hunk 1]
```diff
@@ -48,7 +48,7 @@
...
-  {"LiveFrom": "01/01/2017", "CaseTypeID": "GrantOfRepresentation", "ID": "boFailQA", "Name": "Fail QA", "Description": "Fail QA and mark the case as stopped", "DisplayOrder": 47, "PreConditionState(s)": "BOCaseQA", "PostConditionState": "BOCaseStopped", "CallBackURLAboutToSubmitEvent": "http://${CCD_DEF_CASE_SERVICE_BASE_URL}/tasklist/update", "SecurityClassification": "Public", "ShowSummary": "N"},
+  {"LiveFrom": "01/01/2017", "CaseTypeID": "GrantOfRepresentation", "ID": "boFailQA", "Name": "Fail QA", "Description": "Fail QA and mark the case as stopped", "DisplayOrder": 47, "PreConditionState(s)": "BOCaseQA", "PostConditionState": "BOCaseStopped", "CallBackURLAboutToSubmitEvent": "http://${CCD_DEF_CASE_SERVICE_BASE_URL}/case/failQA", "SecurityClassification": "Public", "ShowSummary": "N"},
```
Comment (sanjayparekh-hmcts) [inline ccdImports/configFiles/CCD_Probate_Backoffice/CaseEvent.json]: prbabaly better to call it fail-qa not failQA

File: src/functionalTest/java/uk/gov/hmcts/probate/functional/caseprogress/SolCcdCaseProgressAdmonWillTests.java [hunk 2]
```diff
@@ -1435,6 +1436,41 @@ public void shouldTransformCaseCorrectlyWhenMarkingReadyToIssue() {
         assertEquals(expected, taskList);
     }
 
+    @Test
+    public void shouldTransformCaseCorrectlyWhenFailQa() {
+        final String response = postCwJson("caseprogressadmonwill/06-caseStopped.json", CASE_FAIL_QA_URL);
+        final JsonPath jsonPath = JsonPath.from(response);
+        final String taskList = jsonPath.get("data.taskList");
+
+        String expectedHtml = "<div class=\"width-50\">\n\n<h2 class=\"govuk-heading-l\">Case progress</h2>\n\n"
```
Comment (sanjayparekh-hmcts) [inline src/functionalTest/java/uk/gov/hmcts/probate/functional/caseprogress/SolCcdCaseProgressAdmonWillTests.java]: you should put this expected response into some resource file + read it as in other tests

File: src/functionalTest/java/uk/gov/hmcts/probate/functional/caseprogress/SolCcdCaseProgressIntestacyTests.java [hunk 3]
```diff
@@ -1436,6 +1437,41 @@ public void shouldTransformCaseCorrectlyWhenMarkingReadyToIssue() {
         assertEquals(expected, taskList);
     }
 
+    @Test
+    public void shouldTransformCaseCorrectlyWhenFailQa() {
+        final String response = postCwJson("caseprogressintestacy/06-caseStopped.json", CASE_FAIL_QA_URL);
+        final JsonPath jsonPath = JsonPath.from(response);
+        final String taskList = jsonPath.get("data.taskList");
+
+        String expectedHtml = "<div class=\"width-50\">\n\n<h2 class=\"govuk-heading-l\">Case progress</h2>\n\n"
```
Comment (sanjayparekh-hmcts) [inline src/functionalTest/java/uk/gov/hmcts/probate/functional/caseprogress/SolCcdCaseProgressIntestacyTests.java]: put response in resource file

File: src/functionalTest/java/uk/gov/hmcts/probate/functional/caseprogress/SolCcdCaseProgressTests.java [hunk 4]
```diff
@@ -1435,6 +1436,41 @@ public void shouldTransformCaseCorrectlyWhenMarkingReadyToIssue() {
         assertEquals(expected, taskList);
     }
 
+    @Test
+    public void shouldTransformCaseCorrectlyWhenFailQa() {
+        final String response = postCwJson("caseprogress/06-caseStopped.json", CASE_FAIL_QA_URL);
+        final JsonPath jsonPath = JsonPath.from(response);
+        final String taskList = jsonPath.get("data.taskList");
+
+        String expectedHtml = "<div class=\"width-50\">\n\n<h2 class=\"govuk-heading-l\">Case progress</h2>\n\n"
```
Comment (sanjayparekh-hmcts) [inline src/functionalTest/java/uk/gov/hmcts/probate/functional/caseprogress/SolCcdCaseProgressTests.java]: same here

File: src/main/java/uk/gov/hmcts/probate/controller/BusinessValidationController.java [hunk 5]
```diff
@@ -227,6 +227,13 @@
         return ResponseEntity.ok(response);
     }
 
+    @PostMapping(path = "/failQA", consumes = APPLICATION_JSON_VALUE, produces = {APPLICATION_JSON_VALUE})
```
Comment (sanjayparekh-hmcts) [inline src/main/java/uk/gov/hmcts/probate/controller/BusinessValidationController.java]: fail-qa

**Review by sanjayparekh-hmcts — CHANGES_REQUESTED** [review]

**Review by sanjayparekh-hmcts — APPROVED** [review]

**Review by isha2040 — APPROVED** [review]

**Pull request merged**